src/lib/action.rs

```diff
@@ -174,7 +174,10 @@ impl RenderJson for Insert {
         json.push(" ".repeat(indent) + "{");
         json.push(format!("{}\"action\": \"insert\",", ind_s));
         json.push(format!("{}\"tree\": {},", &ind_s, self.node.id()));
-        json.push(format!("{}\"parent\": {},", &ind_s, self.new_parent.id()));
+        match self.new_parent {
```

Comment:
Would this be simpler as something like:

```rust
json.push(format!("{}\"parent\": {},", &ind_s, self.new_parent.map_or(|x| x.id(), "")));
```

?

Reply:
Interesting. This doesn't compile, I think because you are saying "either give me a usize, or a str"...

Reply:
I did say "something like" ;) You can convert the usize to a string easily enough.